grpc/params.go

```diff
@@ -84,6 +66,43 @@ func newCallParams(vu modules.VU, input goja.Value) (*callParams, error) {
 	return result, nil
 }
 
+// newMetadata constructs a metadata.MD from the input value.
+func newMetadata(input goja.Value) (metadata.MD, error) {
+	md := metadata.New(nil)
+
+	if input == nil || goja.IsUndefined(input) || goja.IsNull(input) {
+		return md, nil
+	}
+
+	v := input.Export()
+
+	rawHeaders, ok := v.(map[string]interface{})
+	if !ok {
+		return md, errors.New("metadata must be an object with key-value pairs")
+	}
+
+	for hk, kv := range rawHeaders {
+		var val string
+		// The gRPC spec defines that Binary-valued keys end in -bin
+		// https://grpc.io/docs/what-is-grpc/core-concepts/#metadata
```

Comment:
I think we should log it directly in the error message (excluding the link), so we can drop the comment here and have a better error message at the same time.

Reply:
I don't think the suggestion could improve the error message and generally fit the context of the log error message.

I prefer to stick with your [original suggestion, where the comment serves just as for contributors/maintainers and explains the difference in treatment keys with postfixes](https://github.com/grafana/k6/pull/3234#discussion_r1293271020), basically explaining why this `if` exists.

But for sure, we should explain this difference in the documentation, which we will probably do in an upcoming docs PR.

Reply:
>I don't think the suggestion could improve the error message and generally fit the context of the log error message.

It gives you concrete feedback/suggestions about how to fix the issue. If a user reads the docs then I don't expect they will hit the issue, if they are hitting the issue probably they aren't aware of it.

This suggestion is still aligned with the original proposal, but I would extend it to an eventual user hitting the issue. We may preserve the link as a comment if we need more info as developers.

Anyway, this is just a suggestion, feel free to discard it.

Reply:
>It gives you concrete feedback/suggestions about how to fix the issue. 

But that's my point that it's not. Moreover, it could easily confuse the person if we give a link to the GRPC documentation where no k6's specific (like how to pass the binary data) is mentioned.

The error that came from this method is always about the type, but reading the `https://grpc.io/docs/what-is-grpc/core-concepts/#metadata` doesn't always answer, for instance:

```
metadata: {foo: 1}
```

So yeah, sorry, but I'm going to keep it as it is right now.